lib/rules/dot-location.js

```diff
@@ -44,14 +46,30 @@ module.exports = {
          */
         function checkDotLocation(obj, prop, node) {
             const dot = sourceCode.getTokenBefore(prop);
+            const textBeforeDot = sourceCode.getText().slice(obj.range[1], dot.range[0]);
+            const textAfterDot = sourceCode.getText().slice(dot.range[1], prop.range[0]);
 
             if (dot.type === "Punctuator" && dot.value === ".") {
                 if (onObject) {
                     if (!astUtils.isTokenOnSameLine(obj, dot)) {
-                        context.report(node, dot.loc.start, "Expected dot to be on same line as object.");
+                        const neededTextAfterObj = obj.type === "Literal" && typeof obj.value === "number" && /^\d*$/.test(obj.raw)
```

Comment:
A note to me: `/^\d*$/` will be replaced by `astUtils.isDecimalInteger` after #7185 is merged.